### PR TITLE
[Storage] File SDK rename

### PR DESCRIPTION
**Pull request by annatisch**
Fix for issue #8140

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared/models.py [hunk 1]
```diff
@@ -333,17 +333,17 @@ class Services(object):
         Access for the `~azure.storage.blob.BlobServiceClient`
     :param bool queue:
         Access for the `~azure.storage.queue.QueueServiceClient`
-    :param bool file:
-        Access for the `~azure.storage.file.FileServiceClient`
+    :param bool fileshare:
+        Access for the `~azure.storage.fileshare.ShareServiceClient`
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared/models.py]: Should the changes in this shared file be carried over to blobs and queues?

Comment (annatisch) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared/models.py]: Yes - they should :)

File: README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ The libraries released in July preview:
 - [azure-keyvault-secrets](./sdk/keyvault/azure-keyvault-secrets)
 - [azure-identity](./sdk/identity/azure-identity)
 - [azure-storage-blob](./sdk/storage/azure-storage-blob)
-- [azure-storage-file](./sdk/storage/azure-storage-file)
+- [azure-storage-file](./sdk/storage/azure-storage-file-share)
```
Comment (lmazuel) [inline README.md]: Should the link text be changed too?

File: sdk/storage/azure-storage-file-share/HISTORY.md [hunk 3]
```diff
@@ -2,6 +2,15 @@
 
 ## Version 12.0.0b5:
 
+**Important: This package has been renamed**
```
Comment (lmazuel) [inline sdk/storage/azure-storage-file-share/HISTORY.md]: Technically, people that will read this readme will read the one of "azure-storage-file-share". So I would rephrase to something like "This package is the renaming of azure-storage-file" or something like that.

File: README.md (hunk 2 above)
Comment (annatisch) [reply README.md]: I don't think so - this section in the readme is for the historical (July preview) `azure-storage-files` release. I just didn't want the link to point to a 404 so I redirected it to the new equivalent.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py [hunk 4]
```diff
@@ -78,9 +78,10 @@ def __init__(
         self._hosts = kwargs.get("_hosts")
         self.scheme = parsed_url.scheme
 
-        if service not in ["blob", "queue", "file"]:
+        if service not in ["blob", "queue", "file-share"]:
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: wow! we also renamed the endpoint?

Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: Haha it's not rename  I saw you parsed it in the next line

Comment (annatisch) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: Haha - yes we did this because the new package name is used to pass into the pipeline policies on line 105 :)

Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py]: ooh!! Thanks for this info. Do you know why we need to pass package name when we create a pipeline?
`self._create_pipeline(self.credential, storage_sdk=service, **kwargs)`

**Comment by annatisch** [comment]
@scbedd and @lmazuel - I would really appreciate your eyes over this PR given that it will be a new package :)

File: sdk/storage/azure-storage-file-share/README.md [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 
 * Replace or supplement on-premises file servers
 * "Lift and shift" applications
 * Simplify cloud development with shared application settings, diagnostic share, and Dev/Test/Debug tools
 
-[Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage-file/azure/storage/file) | [Package (PyPI)](https://pypi.org/project/azure-storage-file/) | [API reference documentation](https://docs.microsoft.com/en-us/python/api/azure-storage-file/azure.storage.file) | [Product documentation](https://docs.microsoft.com/azure/storage/) | [Samples](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage-file/tests)
+[Source code](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage-file-share/azure/storage/fileshare) | [Package (PyPI)](https://pypi.org/project/azure-storage-file-share/) | [API reference documentation](https://docs.microsoft.com/en-us/python/api/azure-storage-file-share/azure.storage.fileshare) | [Product documentation](https://docs.microsoft.com/azure/storage/) | [Samples](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/storage/azure-storage-file-share/tests)
```
Comment (johanste) [inline sdk/storage/azure-storage-file-share/README.md]: This will conflict with the move of samples to the samples folder.

File: sdk/storage/azure-storage-file-share/README.md [hunk 6]
```diff
@@ -39,25 +39,25 @@ az storage account create -n my-storage-account-name -g my-resource-group
...
 #### Looking up the endpoint URL
-You can find the storage account's file service endpoint URL using the
+You can find the storage account's service endpoint URL using the
 [Azure Portal](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview#storage-account-endpoints),
 [Azure PowerShell](https://docs.microsoft.com/en-us/powershell/module/az.storage/get-azstorageaccount),
 or [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/storage/account?view=azure-cli-latest#az-storage-account-show):
 
 ```bash
-# Get the file service endpoint for the storage account
+# Get the service endpoint for the storage account
```
Comment (johanste) [inline sdk/storage/azure-storage-file-share/README.md]: Get the file service endpoint?

File: sdk/storage/azure-storage-file-share/README.md [hunk 7]
```diff
@@ -75,47 +75,47 @@ connection string instead of providing the account URL and credential separately
...
-4. **[FileClient](https://docs.microsoft.com/en-us/python/api/azure-storage-file/azure.storage.file.fileclient)** -
+4. **[ShareFileClient](https://docs.microsoft.com/en-us/python/api/azure-storage-file-share/azure.storage.fileshare.sharefileclient)** -
     this client represents interaction with a specific file (which need not exist yet). It provides operations to
     upload, download, create, delete, and copy a file.
 
 For details on path naming restrictions, see [Naming and Referencing Shares, Directories, Files, and Metadata](https://docs.microsoft.com/rest/api/storageservices/naming-and-referencing-shares--directories--files--and-metadata).
 
 ## Examples
-The following sections provide several code snippets covering some of the most common Storage File tasks, including:
+The following sections provide several code snippets covering some of the most common Storage File Share tasks, including:
```
Comment (johanste) [inline sdk/storage/azure-storage-file-share/README.md]: Should the description of the examples include the term "Share"? Or should we stay true to the service name? What are the other languages doing here?

File: sdk/storage/azure-storage-file-share/README.md (hunk 5 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-share/README.md]: Pre-emptively updated :)

File: sdk/storage/azure-storage-file-share/README.md (hunk 6 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-share/README.md]: Fixed.

File: sdk/storage/azure-storage-file-share/README.md (hunk 7 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-share/README.md]: I've tried to insert the name "Share" when discussing the service, to try and connect with the new "ShareServiceClient" naming, which doesn't container the term "File" at all....

Will ask around re what other languages are doing.

**Comment by xiafu-msft** [comment]
Give me one minute to double check the name again ^_^

File: sdk/storage/azure-storage-file-share/README.md (hunk 7 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file-share/README.md]: Confirmed that in the portal this is now referred to as "file shares" service.
